src/features/tokenGatedRooms/sagas/create.saga.ts

```diff
@@ -0,0 +1,170 @@
+// get relevant policyFactory
+// deploy with given params
+// assign permissions to the policy using the stream
+
+import { takeEvery } from 'redux-saga/effects'
+import { StreamPermission } from 'streamr-client'
+import { Contract, providers, BigNumber } from 'ethers'
+import { Address } from '$/types'
+import { TokenGatedRoomAction } from '..'
+import handleError from '$/utils/handleError'
+import { RoomId } from '$/features/room/types'
+import setMultiplePermissions from '$/utils/setMultiplePermissions'
+
+import * as ERC20PolicyFactory from '$/contracts/Factories/ERC20PolicyFactory.sol/ERC20PolicyFactory.json'
+import * as ERC721PolicyFactory from '$/contracts/Factories/ERC721PolicyFactory.sol/ERC721PolicyFactory.json'
+import * as ERC777PolicyFactory from '$/contracts/Factories/ERC777PolicyFactory.sol/ERC777PolicyFactory.json'
+import * as ERC1155PolicyFactory from '$/contracts/Factories/ERC1155PolicyFactory.sol/ERC1155PolicyFactory.json'
+
+import { TokenTypes } from '$/features/tokenGatedRooms/types'
+import {
+    ERC1155PolicyFactoryAddress,
+    ERC20PolicyFactoryAddress,
+    ERC721PolicyFactoryAddress,
+    ERC777PolicyFactoryAddress,
+    PermissionType,
+} from '$/features/tokenGatedRooms/utils/const'
+import { getPolicyRegistry } from '$/features/tokenGatedRooms/utils/getPolicyRegistry'
+import toast, { Controller } from '$/features/toaster/helpers/toast'
+import { ToastType } from '$/components/Toast'
+
+async function waitForPolicyToBeDeployed(
+    registry: Contract,
+    tokenAddress: Address,
+    tokenId: BigNumber,
+    roomId: RoomId,
+    stakingEnabled: boolean
+): Promise<Address> {
+    const policyAddress: Address = await registry.getPolicy(
+        tokenAddress,
+        tokenId,
+        roomId,
+        stakingEnabled
+    )
+
+    if (policyAddress === '0x0000000000000000000000000000000000000000') {
+        console.warn(
+            'policy at 0x0000000000000000000000000000000000000000 address, waiting 1 second and trying again'
+        )
+        await new Promise((resolve) => setTimeout(resolve, 1000))
+        return waitForPolicyToBeDeployed(registry, tokenAddress, tokenId, roomId, stakingEnabled)
+    }
+
+    return policyAddress
+}
+
+function* onCreate({
+    payload: {
+        owner,
+        tokenAddress,
+        tokenType,
+        roomId,
+        streamrClient,
+        minRequiredBalance,
+        tokenIds,
+        stakingEnabled,
+        provider,
+    },
+}: ReturnType<typeof TokenGatedRoomAction.create>) {
+    let tc: Controller | undefined
+    try {
+        tc = yield toast({
+            title: 'Deploying Token Gate',
+            type: ToastType.Info,
```

Comment:
This could be `ToastType.Processing` but it won't go away, ever.

Have a look at `retoast`, and how I use it in other flows. The situation with `Controller` is a bit mixed up because there are 2. One reduced (see `toast`, no `open`) and another complete (w/ `open`). See `Toaster`. I'll get that sorted oot soon.